docs/source/examples/elasticsearch_outlet.rst

```diff
@@ -0,0 +1,85 @@
+ElasticSearch Outlet
+--------------------
+
+.. container:: tutorial-block
+
+    In this example we create an implementation of :any:`Outlet` that indexes documents to a running Elasticsearch instance. 
+    
+    **Note**: this example assumes that Elasticsearch is correctly configured and that the index you are indexing documents to exists
+    with the appropriate mappings.  
+
+    #. Extend the :any:`Outlet` with new parameters required when constructing: :code:`es_client` an instance of the elasticsearch python client and :code:`index_name` the name of a pre-existing index in the running cluster.
+
+    .. rst-class:: highlight-small
+    .. literalinclude:: ../../examples/elasticsearch_outlet.py
+        :language: python
+        :start-at: class ElasticSearchIndexerOutlet
+        :end-at: raise RuntimeError(f"Index '{self.index_name}' does not exist ")
+
+    #. In this implementation of the :code:`push` method there are a few custom behaviors specified. As we iterate over every incoming record:
+
+        * We use the dict keys from the current record's :code:`payload` as our unique document id.
```

Comment:
ID would be in [capitals rather than lowercase too.](https://english.stackexchange.com/questions/101248/how-should-the-abbreviation-for-identifier-be-capitalized.)